Network/AMQP/Internal.hs

```diff
@@ -518,7 +531,7 @@ channelReceiver chan = do
             )
     handleAsync (SimpleMethod (Channel_close _ (ShortString errorMsg) _ _)) = do
         closeChannel' chan errorMsg
-        killThread =<< myThreadId
+        myThreadId >>= flip CE.throwTo (ConnectionClosedException . T.unpack $ errorMsg)
```

Comment:
Maybe, but throwIO doesn't allow you to target a thread, so this line is closer in its spirit to the original code. Any compelling reason not to use throwTo?